server/datastore/s3/software_installer.go

```diff
@@ -74,6 +74,53 @@ func (i *SoftwareInstallerStore) Exists(ctx context.Context, installerID string)
 	return true, nil
 }
 
+func (i *SoftwareInstallerStore) Cleanup(ctx context.Context, usedInstallerIDs []string) (int, error) {
```

Comment:
as a super minor optimization we could return early if the list is empty (because I believe this method will be frequently called without any ids)

Reply:
Oh, great point, yeah I'll add that!

Reply:
Oh, no sorry, if the list is empty it means there are no **existing and in use** software installers, not that there's nothing to do. There may be a lot of unused stuff to remove.

Reply:
you're totally right, it's the opposite!